[BIOMAGE-574] Create redux store for samples

xverges opened the pull request:
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-574

#### Link to staging deployment URL 

N/A

#### Links to any Pull Requests related to this

#136 does quite the same, but for projects. A simple merge conflict will need to be fixed to merge both.

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

* implemented actions, actionTypes and reducers for `samples`
* no running code uses the store

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR

Felipe, the old dog that we adopted a month ago.

![felipe](https://user-images.githubusercontent.com/460418/110095021-bdf12600-7d9c-11eb-8996-fe5ceda66ed5.jpeg)

xverges committed:
Lint README.md

xverges committed:
Create redux store for samples

xverges commented on src/__test__/redux/reducers/samplesReducer.test.js:
```diff
@@ -0,0 +1,90 @@
+      'features.tsv': {
+        ...sampleFileTemplate,
+        name: 'features.tsv',
+      },
```
I am not sure if adding the files to a test sample is of any use, because it not verified by any test. The same applies to creating the `sampleFileTemplate` in the reducer's initial state.

xverges commented on README.md:
```diff
@@ -1,5 +1,6 @@
 UI
```
Linted markdown, because it was annoying me in Visual Studio Code's error pane (I have https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint)

Is it acceptable to do small changes unrelated to the main PR goal?

aerlaut replied:
No problem 👍

xverges commented on src/redux/reducers/index.js:
```diff
@@ -10,16 +10,18 @@ import componentConfigReducer from './componentConfig/index';
+import sampleReducer from './samples';
 
 export default combineReducers({
+  cellInfo: cellInfoReducer,
```
Sorted the reducers alphabetically.

xverges committed:
Fix copypaste errors

xverges committed:
Use proper uuid + remove unnecesary snapshot

aerlaut commented on src/redux/reducers/samples/initialState.js:
```diff
@@ -0,0 +1,26 @@
+  complete: false,
+  error: false,
+  fileNames: [],
+  files: {},
```
can you add a `createdDate` and `lastModified` field? These fields should store `moment().isoTime()`

aerlaut commented on src/redux/reducers/samples/initialState.js:
```diff
@@ -0,0 +1,26 @@
+  size: 0,
+  mime: '',
+  success: false,
+  error: false,
```
Is `error` to store error messages? If so, maybe it's better to set it's default value to an empty string?

xverges replied:
> error : false if there is no error associated with sample. Else, contain array of error messages. Default : false

I just verified if an empty array was falsy, but it is not

marcellp replied:
I agree with Xavier's choice here, having error to be false and a string if there is an error is actually the common practice in Redux and it also lends itself to easier coding styles.

xverges replied:
Not _Xavier's choice_: it is what Agi specified in Jira, although he forgot when making this comment :-)
(And now I realize that my previous comment intended to quote the original ticket 🤔 )

xverges committed:
Merge master into create-redux-store-for-samples

xverges committed:
Add created/modified timestamps to sample redux

* Added also a new test for project actions
* :-( wasted a considerable amount of time wrongly trying to test the
  update action in the reducer test file. Also failed to mock
  moment().toISOString()

marcellp reviewed (approved):
lgtm
